nzedb/Forum.php

```diff
@@ -23,119 +25,175 @@ public function __construct(array $options = [])
+	 */
 	public function getPosts($parent)
 	{
 		return $this->pdo->query(
-			sprintf("
+			sprintf('
 				SELECT forum_posts.*, users.username
 				FROM forum_posts
 				LEFT OUTER JOIN users ON users.id = forum_posts.user_id
 				WHERE forum_posts.id = %d OR parentid = %d
 				ORDER BY createddate ASC
-				LIMIT 250",
+				LIMIT 250',
 				$parent,
 				$parent
 			)
 		);
 	}
 
+	/**
+	 * @param $id
+	 *
+	 * @return array|bool
+	 */
 	public function getPost($id)
 	{
-		return $this->pdo->queryOneRow(sprintf("SELECT * FROM forum_posts WHERE id = %d", $id));
+		return $this->pdo->queryOneRow(sprintf('SELECT * FROM forum_posts WHERE id = %d', $id));
 	}
 
+	/**
+	 * @return int
+	 */
 	public function getBrowseCount()
 	{
-		$res = $this->pdo->queryOneRow(sprintf("SELECT COUNT(id) AS num FROM forum_posts WHERE parentid = 0"));
-		return ($res === false ? 0 : $res["num"]);
+		$res = $this->pdo->queryOneRow(sprintf('SELECT COUNT(id) AS num FROM forum_posts WHERE parentid = 0'));
+		return ($res === false ? 0 : $res['num']);
 	}
 
+	/**
+	 * @param $start
+	 * @param $num
+	 *
+	 * @return array
+	 */
 	public function getBrowseRange($start, $num)
 	{
 		return $this->pdo->query(
-			sprintf("
+			sprintf('
 				SELECT forum_posts.*, users.username
 				FROM forum_posts
 				LEFT OUTER JOIN users ON users.id = forum_posts.user_id
 				WHERE parentid = 0
-				ORDER BY updateddate DESC %s",
-				($start === false ? '' : (" LIMIT " . $num . " OFFSET " . $start))
+				ORDER BY updateddate DESC %s',
+				$start === false ? '' : ' LIMIT ' . $num . ' OFFSET ' . $start
 			)
 		);
 	}
 
+	/**
+	 * @param $parent
+	 */
 	public function deleteParent($parent)
 	{
-		$this->pdo->queryExec(sprintf("DELETE FROM forum_posts WHERE id = %d OR parentid = %d", $parent, $parent));
+		$this->pdo->queryExec(sprintf('DELETE FROM forum_posts WHERE id = %d OR parentid = %d', $parent, $parent));
 	}
 
+	/**
+	 * @param $id
+	 */
 	public function deletePost($id)
 	{
 		$post = $this->getPost($id);
 		if ($post) {
-			if ($post["parentid"] == "0") {
+			if ((int) $post['parentid'] === 0) {
 				$this->deleteParent($id);
 			} else {
-				$this->pdo->queryExec(sprintf("DELETE FROM forum_posts WHERE id = %d", $id));
+				$this->pdo->queryExec(sprintf('DELETE FROM forum_posts WHERE id = %d', $id));
 			}
 		}
 	}
 
+	/**
+	 * @param $id
+	 */
 	public function deleteUser($id)
 	{
-		$this->pdo->queryExec(sprintf("DELETE FROM forum_posts WHERE user_id = %d", $id));
+		$this->pdo->queryExec(sprintf('DELETE FROM forum_posts WHERE user_id = %d', $id));
 	}
 
+	/**
+	 * @param $uid
+	 *
+	 * @return int
+	 */
 	public function getCountForUser($uid)
 	{
-		$res = $this->pdo->queryOneRow(sprintf("SELECT COUNT(id) AS num FROM forum_posts WHERE user_id = %d", $uid));
-		return ($res === false ? 0 : $res["num"]);
+		$res = $this->pdo->queryOneRow(sprintf('SELECT COUNT(id) AS num FROM forum_posts WHERE user_id = %d', $uid));
+		return ($res === false ? 0 : $res['num']);
```

Comment:
cast $res['num'] as it might be a string and the method is documented as returning an int.